crates/storage/traits/src/storage.rs

```diff
@@ -44,6 +44,30 @@ where
     }
 }
 
+/// Extension trait to [`Storable`] that ensures that the given input data is consumed
+/// completely.
+pub trait StorableDecodeAll: Sized {
+    /// Decode `Self` and consume all of the given input data.
+    ///
+    /// If not all data is consumed, an error is returned.
+    fn decode_all(input: &mut &[u8]) -> Result<Self, scale::Error>;
+}
+
+impl<P> StorableDecodeAll for P
+where
+    P: Storable,
+{
+    fn decode_all(input: &mut &[u8]) -> Result<Self, scale::Error> {
```

Comment:
Why do we want a separate `StorableDecodeAll` trait with blanket implementation when we can use this code directly in the `get_contract_storage` and `take_contract_storage`? (If you want to reuse the code, then you can have a procedure method)

Reply:
Yes, that might be a better solution instead of the extra trait :+1:

Reply:
Done